extensions/reviewed/TopDownMovementAnimator.json

```diff
@@ -0,0 +1,898 @@
+                    "Object",
+                    "Behavior",
+                    "=",
+                    "mod(mod(round(GetArgumentAsNumber(\"Value\")), Object.Behavior::PropertyDirectionCount()) + Object.Behavior::PropertyDirectionCount(), Object.Behavior::PropertyDirectionCount())"
+                  ]
+                },
+                {
+                  "type": {
+                    "value": "TopDownMovementAnimator::TopDownMovementAnimator::SetSharedPropertyAnimationFrame"
+                  },
+                  "parameters": [
+                    "Object",
+                    "Behavior",
+                    "=",
+                    "Object.Sprite()"
+                  ]
+                },
+                {
+                  "type": {
+                    "value": "SetAnimationName"
+                  },
+                  "parameters": [
+                    "Object",
+                    "Object. Behavior::PropertyAnimationName() + ToString(Object.Behavior::PropertyAnimationDirection())"
+                  ]
+                },
+                {
+                  "type": {
+                    "value": "ChangeSprite"
+                  },
+                  "parameters": [
+                    "Object",
+                    "=",
+                    "Object.Behavior::SharedPropertyAnimationFrame()"
+                  ]
+                }
+              ]
+            }
+          ],
+          "parameters": [
+            {
+              "description": "Object",
+              "name": "Object",
+              "supplementaryInformation": "Sprite",
+              "type": "object"
+            },
+            {
+              "description": "Behavior",
+              "name": "Behavior",
+              "supplementaryInformation": "TopDownMovementAnimator::TopDownMovementAnimator",
+              "type": "behavior"
+            },
+            {
+              "description": "Value",
+              "name": "Value",
+              "type": "expression"
+            }
+          ],
+          "objectGroups": []
+        }
+      ],
+      "propertyDescriptors": [
+        {
+          "value": "",
+          "type": "Behavior",
+          "label": "Top-down movement",
+          "description": "",
+          "group": "",
+          "extraInformation": [
+            "TopDownMovementBehavior::TopDownMovementBehavior"
+          ],
+          "hidden": false,
+          "name": "TopDownMovement"
+        },
+        {
+          "value": "true",
+          "type": "Boolean",
+          "label": "Scale animation according to speed",
+          "description": "",
+          "group": "Animation",
+          "extraInformation": [],
+          "hidden": false,
+          "name": "IsScalingAnimation"
+        },
+        {
+          "value": "Walk",
+          "type": "String",
+          "label": "Animation name",
+          "description": "Animations must be called \"Walk0\", \"Walk1\"... for left, down...",
```

Comment:
Could we add more details here? I think it has to be exhaustive to be really helpful

Reply:
It gives the start and the direction. I think giving more won't help. Especially because there are not necessarily only 4 directions.